### PR TITLE
Lvnod 47 home page

### PR DESCRIPTION
**Pull request by AlbaMontis**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-46)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-47)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-48)

## Code reviewers

- [x] @denisbodnar
- [x] @shyyka
- [x] @yuliannakorol

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

### CODE REVIEW

**Review by denisbodnar — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

File: app/src/components/CheckEmail/CheckEmail.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+export default class CheckEmail extends React.Component {
```
Comment (4asof4ik) [inline app/src/components/CheckEmail/CheckEmail.js]: this component should be common for all messages

File: app/src/components/Home/Home.js [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+      <Grid>
+        <Row className="show-grid">
+          <Col sm={12} md={8}>
+            <PageHeader className="center">Da-Mi</PageHeader>
+            {this.props.children}
+            <p id="homeText">Some text about our project!!!</p>
+          </Col>
+          <Col sm={12} md={4}>
+            <Login />
+            <p className="center">OR</p>
```
Comment (4asof4ik) [inline app/src/components/Home/Home.js]: bootstrap has own class text-center which align text

File: app/src/components/Home/Home.js [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
 export default class Home extends React.Component {
   render() {
-    return <h1>Hello here, its alive!</h1>;
+    return (
+      <Grid>
+        <Row className="show-grid">
+          <Col sm={12} md={8}>
+            <PageHeader className="center">Da-Mi</PageHeader>
+            {this.props.children}
+            <p id="homeText">Some text about our project!!!</p>
```
Comment (4asof4ik) [inline app/src/components/Home/Home.js]: Why did you add id here?

File: app/src/components/Home/Home.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.center {
```
Comment (4asof4ik) [inline app/src/components/Home/Home.scss]: drop this style

File: webpack/webpack.config.dev.js [hunk 5]
```diff
@@ -37,7 +37,10 @@ module.exports = {
       filename: 'index.html',
       inject: false,
       template: require('html-webpack-template'),
-      appMountId: 'root'
+      appMountId: 'root',
+      links: [
```
Comment (4asof4ik) [inline webpack/webpack.config.dev.js]: this is bad idea

File: app/src/components/Main/Main.js [hunk 6]
```diff
@@ -1,14 +1,25 @@
 import React from 'react';
-import { Link } from 'react-router';
+import { Navbar, Nav, NavItem } from 'react-bootstrap';
+import { LinkContainer } from 'react-router-bootstrap';
+import style from './Main.scss';
 
 export default class Main extends React.Component {
```
Comment (4asof4ik) [inline app/src/components/Main/Main.js]: rename into Navigations or something else

File: app/src/router.js [hunk 7]
```diff
@@ -1,12 +1,17 @@
 import React from 'react';
 import { Route, IndexRoute } from 'react-router';
-import { About, Home, Contacts, Main } from './components';
+import { About, Home, Contacts, Signup, Main, CheckEmail}
```
Comment (4asof4ik) [inline app/src/router.js]: space before }

File: app/src/router.js [hunk 8]
```diff
@@ -1,12 +1,17 @@
...
+from './components';
 
 const Routes = (
   <Route path='/' component={Main}>
```
Comment (4asof4ik) [inline app/src/router.js]: 9 line has path "/" too

File: app/src/components/Signup/Signup.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} required/>
+    </FormGroup>
+  );
+}
+
+export default class Signup extends React.Component {
+  render () {
```
Comment (4asof4ik) [inline app/src/components/Signup/Signup.js]: read about react state

File: app/src/components/Signup/Signup.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+            type="password"
+            placeholder="Enter your password"
+          />
+          <FieldGroup
+            id="password-repeat"
+            label="REPEAT YOUR PASSWORD"
+            type="password"
+            placeholder="Repeat your password"
+          />
+          <Link to='/email'>
```
Comment (4asof4ik) [inline app/src/components/Signup/Signup.js]: after click should redirect to login page and show message

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: app/src/components/Login/Login.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+export default class Login extends React.Component {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      email: '',
+      password: ''
+    };
+
+    this.handleChange = this.handleChange.bind(this);
```
Comment (4asof4ik) [inline app/src/components/Login/Login.js]: create array for function and use .map

File: app/src/components/Login/Login.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    return (
+      <Form horizontal onSubmit={this.handleSubmit}>
+        <FormGroup controlId="email">
+          <Col componentClass={ControlLabel} sm={12} md={3}>
+            Email
+          </Col>
+          <Col sm={12} md={9}>
+            <FormControl type="email"
+                         placeholder="Email"
+                         value={this.state.email} onChange={this.handleChange}
```
Comment (4asof4ik) [inline app/src/components/Login/Login.js]: onChange={value => this.handleValue('stateName', value)

File: app/src/components/Login/Login.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    this.state = {
+      email: '',
+      password: ''
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
```
Comment (4asof4ik) [inline app/src/components/Login/Login.js]: handleChange(state, event)

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

File: app/src/components/Login/Login.js [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+  }
+
+  render () {
+    return (
+      <Form horizontal onSubmit={this.handleSubmit}>
+        <FormGroup controlId="email">
```
Comment (4asof4ik) [inline app/src/components/Login/Login.js]: Create function which will be returning formgroup for email and password

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: app/src/components/Login/Login.js [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+    this.setState({[stateName]: e.target.value});
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+  }
+
+  render () {
+    return (
+      <Form horizontal onSubmit={this.handleSubmit}>
```
Comment (4asof4ik) [inline app/src/components/Login/Login.js:36]: Form horizontal onSubmit={this.handleSubmit}
  {['email', password].map(param => {
      InputGroup id={param} type={param} .......
})}

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by shyyka — APPROVED** [review]

**Review by denisbodnar — APPROVED** [review]

**Review by yuliannakorol — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**